### PR TITLE
Add `Ipv6Addr::is_benchmarking`

### PR DESCRIPTION
**Pull request by CDirkx**
This PR adds the unstable method `Ipv6Addr::is_benchmarking`. This method is added for parity with `Ipv4Addr::is_benchmarking`, and I intend to use it in a future rework of `Ipv6Addr::is_global` (edit: #86634) to more accurately follow the [IANA Special Address Registry](https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml) (like is done in `Ipv4Addr::is_global`).

With `Ipv6Addr::is_benchmarking` and `Ipv4Addr::is_benchmarking` now both existing, `IpAddr::is_benchmarking` is also added.

### CODE REVIEW

File: library/std/src/net/ip/tests.rs [hunk 1]
```diff
@@ -475,21 +484,22 @@ fn ipv6_properties() {
...
+            let documentation: u32 = 1 << 8;
+            let benchmarking: u32 = 1 << 16;
+            let multicast_interface_local: u32 = 1 << 9;
+            let multicast_link_local: u32 = 1 << 10;
+            let multicast_realm_local: u32 = 1 << 11;
+            let multicast_admin_local: u32 = 1 << 12;
+            let multicast_site_local: u32 = 1 << 13;
+            let multicast_organization_local: u32 = 1 << 14;
+            let multicast_global: u32 = 1 << 15;
+            let multicast: u32 = multicast_interface_local
```
Comment (CDirkx) [inline library/std/src/net/ip/tests.rs:502]: More than 16 properties are being checked, so the tag type is changed from an `u16` to an `u32`.

File: library/std/src/net/ip/tests.rs [hunk 2]
```diff
@@ -313,6 +321,7 @@ fn ip_properties() {
     check!("ff08::", multicast);
     check!("ff0e::", global | multicast);
     check!("2001:db8:85a3::8a2e:370:7334", doc);
+    check!("2001:2::ac32:23ff:21", global | benchmarking);
```
Comment (CDirkx) [inline library/std/src/net/ip/tests.rs:324]: According to the [IANA IPv6 Special Address Registry](https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml) benchmarking addresses are not globally reachable. However due to the current simplistic implementation of `Ipv6Addr::is_global` compared to `Ipv4Addr::is_global`, Rust currently considers benchmarking addresses global. I plan to address this in a future PR reworking `is_global`.

File: library/std/src/net/ip/tests.rs [hunk 3]
```diff
@@ -679,6 +695,12 @@ fn ipv6_properties() {
         documentation
     );
 
+    check!(
+        "2001:2::ac32:23ff:21",
+        &[0x20, 1, 0, 2, 0, 0, 0, 0, 0, 0, 0xac, 0x32, 0x23, 0xff, 0, 0x21],
+        global | unicast_global | benchmarking
```
Comment (CDirkx) [inline library/std/src/net/ip/tests.rs:701]: Same as above, benchmarking addresses should not actually be globally reachable. They do however have global unicast scope.

File: library/std/src/net/ip/tests.rs (hunk 2 above)
Comment (CDirkx) [reply library/std/src/net/ip/tests.rs:324]: The rework of `Ipv6Addr::is_global` has been submitted (#86634).

**Comment by camelid** [comment]
r? rust-lang/libs-api

**Comment by camelid** [comment]
r? rust-lang/libs

**Comment by camelid** [comment]
r? libs

**Pull request merged**